libsolidity/analysis/SyntaxChecker.cpp

```diff
@@ -226,6 +226,22 @@ bool SyntaxChecker::visit(FunctionDefinition const& _function)
 				"\"."
 			);
 	}
+
+	if (_function.isOldStyleConstructor())
+	{
+		if (v050)
+			m_errorReporter.syntaxError(
+				_function.location(),
+				"Functions are not allowed to have the same name as the contract. "
+				"For constructors use the \"constructor\" keyword instead."
```

Comment:
It would probably be good to explain it in a little more detail: ``If you intend this to be a constructor, use "constructor(...) { ... }" to define it.`` - although this is probably still not the best solution.

Reply:
Same applies below, of course.